ion_hash_test.go

```diff
@@ -0,0 +1,595 @@
+			err = binaryWriter.FieldName(reader.FieldName())
+			require.NoError(t, err)
+		}
+		if reader.IsNull() {
+			err := textWriter.WriteNullType(ion.BlobType)
+			require.NoError(t, err)
+			err = binaryWriter.WriteNullType(ion.BlobType)
+			require.NoError(t, err)
+		} else {
+			byteValue, err := reader.ByteValue()
+			require.NoError(t, err)
+			err = textWriter.WriteBlob(byteValue)
+			require.NoError(t, err)
+			err = binaryWriter.WriteBlob(byteValue)
+			require.NoError(t, err)
+		}
+
+	case ion.ClobType:
+		if reader.FieldName() != "" {
+			err := textWriter.FieldName(reader.FieldName())
+			require.NoError(t, err)
+			err = binaryWriter.FieldName(reader.FieldName())
+			require.NoError(t, err)
+		}
+		if reader.IsNull() {
+			err := textWriter.WriteNullType(ion.ClobType)
+			require.NoError(t, err)
+			err = binaryWriter.WriteNullType(ion.ClobType)
+			require.NoError(t, err)
+		} else {
+			byteValue, err := reader.ByteValue()
+			require.NoError(t, err)
+			err = textWriter.WriteClob(byteValue)
+			require.NoError(t, err)
+			err = binaryWriter.WriteClob(byteValue)
+			require.NoError(t, err)
+		}
+
+	case ion.DecimalType:
+		if reader.FieldName() != "" {
+			err := textWriter.FieldName(reader.FieldName())
+			require.NoError(t, err)
+			err = binaryWriter.FieldName(reader.FieldName())
+			require.NoError(t, err)
+		}
+		if reader.IsNull() {
+			err := textWriter.WriteNullType(ion.DecimalType)
+			require.NoError(t, err)
+			err = binaryWriter.WriteNullType(ion.DecimalType)
+			require.NoError(t, err)
+		} else {
+			decimalValue, err := reader.DecimalValue()
+			require.NoError(t, err)
+			err = textWriter.WriteDecimal(decimalValue)
+			require.NoError(t, err)
+			err = binaryWriter.WriteDecimal(decimalValue)
+			require.NoError(t, err)
+		}
+
+	case ion.FloatType:
+		if reader.FieldName() != "" {
+			err := textWriter.FieldName(reader.FieldName())
+			require.NoError(t, err)
+			err = binaryWriter.FieldName(reader.FieldName())
+			require.NoError(t, err)
+		}
+		if reader.IsNull() {
+			err := textWriter.WriteNullType(ion.FloatType)
+			require.NoError(t, err)
+			err = binaryWriter.WriteNullType(ion.FloatType)
+			require.NoError(t, err)
+		} else {
+			floatValue, err := reader.FloatValue()
+			require.NoError(t, err)
+			err = textWriter.WriteFloat(floatValue)
+			require.NoError(t, err)
+			err = binaryWriter.WriteFloat(floatValue)
+			require.NoError(t, err)
+		}
+
+	case ion.IntType:
+		if reader.FieldName() != "" {
+			err := textWriter.FieldName(reader.FieldName())
+			require.NoError(t, err)
+			err = binaryWriter.FieldName(reader.FieldName())
+			require.NoError(t, err)
+		}
+		intSize, err := reader.IntSize()
+		require.NoError(t, err)
+
+		if reader.IsNull() {
+			err = textWriter.WriteNullType(ion.IntType)
+			require.NoError(t, err)
+			err = binaryWriter.WriteNullType(ion.IntType)
+			require.NoError(t, err)
+		} else {
+			switch intSize {
+			case ion.Int32:
+
+				intValue, err := reader.IntValue()
```

Comment:
If I recall correctly, when intSize is Int32, we can still call `reader.Int64Value()` to get the 64 bit version of the int value and pass that directly to `WriteInt()` instead of calling `reader.IntValue()` to get the 32 bit value and then converting it to 64 bit when passing to `WriteInt()`.

Because of this, we can combine Int32 and Int64 into one case.

Take a look at the `switch IntSize` in `writeFromReaderToWriter()` in `testing_utilities.go` to see another example of this.

Reply:
Oh wow, I didnt realize writeFromReaderToWriter() existed, thanks Ill take a look

Reply:
Updated